tests/utils.py

```diff
@@ -123,7 +123,11 @@ def apply(self, **kwargs):
         t_end = kwargs['t_end']
         assert(t_start <= t_end)
         if self.direction == 1:
-            self.u[:] = self.u[:] + self.direction * abs(t_start - t_end)
+            for t in range(t_start, t_end):
```

Comment:
What is happening here?

Reply:
We added a new dimension to the `u` array.  This dimension is responsible to save the `block_size` time steps. That before are not saved because isn't necessary.  This loop is responsible to write each time step in the right position.